src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,897 @@
+package duke;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File; 
+import java.io.FileWriter; 
+import java.io.FileReader; 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+public class Duke {
+
+    public static int TODO = 0;
+    public static int DEADLINE = 1;
+    public static int EVENT = 2;
+
+    public static class Ui {
+        /**
+         * Returns a Ui object which is used to output
+         * to the terminal to communicate with user
+         *
+         * @param openingText the opening message sent to users.
+         * @param closingText the final message sent to users.
+         * @return Ui object
+         */
+        Ui(String openingText, String closingText) {
+            this.openingText = openingText;
+            this.closingText = closingText;
+            this.horizontalLine = horizontalLine = "----------------------------------------------------------";
+        }
+
+        /**
+         * Returns void, just prints a horizontal line
+         */
+        void printHorizontalLine() {
```

Comment:
method names follow convention (start with verb, written in camelCase) and are really organized! great work on that